### ISSUE TITLE
Exception : number_format(): Argument #1 ($num) must be of type float, string given

### ISSUE THREAD

**Issue by Profxmas** [amandadoughty/moodle-mod_peerwork#20]
Hi,

We have:
Plugin v4.0
Moodle 4.1.3

When the teacher choose the option for the student to see the comments anonymoulsy, the student got this error.
Previously we were with 3.9 with no problems.
![2024-03-27_13h40_04](https://github.com/amandadoughty/moodle-mod_peerwork/assets/150482645/a4ca23dd-6042-45ac-bd5c-53354959afdc)

More information on this errors conduct to:
https://docs.moodle.org/4x/fr/error/moodle/generalexceptionmessage
![image](https://github.com/amandadoughty/moodle-mod_peerwork/assets/150482645/31c4e345-6c93-4a63-bc50-3b4d6967dc12)

Thanks !

**Comment by barishgolland** [comment]
We're receiving the same error. Is it a PHP8 issue? Anyone know if they'll be updating this plugin anytime soon?

**Comment by manuelmovilla** [comment]
Hi,

By creating a backup of all the Peer Work resources in the course and then restoring this backup within the same course without deleting the prior course information, the issue has been resolved, and students no longer have any problems. It is important to be very careful not to delete the prior course information during this process.

Kind regards,

**Comment by mhughes2k** [comment]
Not sure if this is related, but we've seen this as well and here's the stack trace:

```
Stack trace:
line 8978 of /lib/moodlelib.php: TypeError thrown
line 8978 of /lib/moodlelib.php: call to number_format()
line 246 of /mod/peerwork/classes/details_form.php: call to format_float()
line 197 of /mod/peerwork/details.php: call to mod_peerwork_details_form->set_data()
```

**Comment by mhughes2k** [comment]
the data that appears to be "-" appears to be $data['finalgrades']['revisedgrade'] for a user.

This value appears to have come from a gradebook override which the "string value" appears as "-".

**Comment by amandadoughty** [comment]
@mhughes2k did you have a fix for this?

**Comment by michael-hughes-strath** [comment]
No, turns out I didn't start on one, but I can't reproduce this on the current version.

**Comment by amandadoughty** [comment]
I've got some vague memory of us fixing it.

**Comment by amandadoughty** [comment]
Closing as no longer an issue
